aten/src/ATen/native/Math.h

```diff
@@ -381,6 +381,455 @@ static inline float calc_polygamma(int64_t n, float x) {
       zeta(double(n + 1), x);
 }
 
+// regularized lower incomplete gamma
```

Comment:
If this code is copied or derived from SciPy we should be sure to cite it properly. See, for example. how we cite Cephes in this same file. 

@rgommers I couldn't find the SciPy license in PyTorch, seems like we should add it here?

Reply:
That's one option, but probably not the optimal one.  I had a look at how all the license info is structured, and there's a few issues with it. The `NOTICE` file contains license info of some vendored code. I think the SciPy license should be added there - that will future-proof things so SciPy code can simply be copied in the future, without having to worry about adding more license info each time.

`NOTICE` should also be referenced from `LICENSE`.

There may be more to do, for example `NOTICE` says that Caffe2 is Apache 2 licensed, which is important and should be reflected in the overall PyTorch license that is advertised in for example the PyPI classifier (which is no longer just BSD, it's hybrid and not GPLv2-compatible - which is luckily no longer very important).

Reply:
Looks like `NOTICE` just contains a few Caffe2 licenses? Looks like @ezyang worked on the file (like nearly every file in PyTorch). @ezyang, is that the right place to stuff the SciPy license?